### PR TITLE
Add subdirectories to engine/ similar to pokecrystal

### PR DESCRIPTION
**Pull request by Rangi42**
Still remaining:

- Subdirectories in data/
- Move all code from home.asm to files in home/
- Name `SECTION`s
- Use charmaps
- Maybe split up files like engine/events/hidden_object_functions\*.asm
- Issue #249 (RAM labels)
- Issue #165 (`PSYCHIC_TYPE` like pokecrystal, or `TYPE_` prefix on all constants?)

That's all for future PRs at some later date though.

### CODE REVIEW

File: audio.asm [hunk 1]
```diff
@@ -352,248 +352,27 @@ INCLUDE "audio/sfx/cry21_3.asm"
...
-
-;written to stop the alarm
-.toneDataSilence
-	db $00,$00,$00,$80
-
-
-INCLUDE "engine/menu/bills_pc.asm"
-
+INCLUDE "audio/low_health_alarm.asm"
+INCLUDE "engine/pokemon/bills_pc.asm"
```
Comment (dannye) [inline audio.asm:365]: Now that linkerscripts are a thing, can this include be moved from audio.o to main.o and still link properly?

**Comment by dannye** [comment]
I'm not sure I'd say that `clear_save.asm` and `init_player_data.asm` belong in `engine/movie/` unless you have a specific reason.

You changed `engine/menu/` to `engine/menus/` which makes sense so it is now consistent with the plural `engine/items/` and `engine/events/` for example. But now `engine/movie/` feels inconsistent for being singular.

**Comment by dannye** [comment]
I know I'm the one to blame, but my least favorite files names are what is now `engine/events/hidden_object_functions*.asm` because they were pre-linkerscript.
Do you think now it would be better organized to have just a single `engine/events/hidden_object_functions.asm` file with a section per bank?

File: audio.asm (hunk 1 above)
Comment (Rangi42) [reply audio.asm]: Yes it can.

**Comment by Rangi42** [comment]
I agree about plural "movies", but went with the same name as pokecrystal. @mid-kid originally named it "movie". Can we agree on one name for both? Or if you like "movies" here and pfero likes "movie" there, that's alright; there are other mismatched gen 1/2 filenames.

I'm not sure if a single multi-section hidden_object_functions.asm file would be best, or a hidden_object_functions/ subdirectory that breaks up each single-bank file into multiple files (bookshelves.asm, statues.asm, cinnabar_quiz.asm, etc).

**Comment by dannye** [comment]
"movie" is fine for matching with pokecrystal.

I like your idea of `engine/events/hidden_object_functions/`.

**Comment by Rangi42** [comment]
Yeah, I may do `hidden_object_functions` later. Thanks!

**Pull request merged**